db/osqlblockproc.c

```diff
@@ -1305,6 +1305,18 @@ int bplog_schemachange_wait(struct ireq *iq, int rc)
     iq->sc_pending = NULL;
 
     while (sc != NULL) {
+        if (!sc->nothrevent) {
+            // If sc is running asynchronously, we must first hold mtxStart and
+            // wait on condStart until sc->started is set to true—otherwise we risk
+            // locking sc->mtx before the sc has begun. 
+            Pthread_mutex_lock(&sc->mtxStart);
+            while (!sc->started) {
+                Pthread_cond_wait(&sc->condStart, &sc->mtxStart);
+            }
+            Pthread_mutex_unlock(&sc->mtxStart);
+        }
```

Comment:
This wait is acceptable: it doesn't block the master from coming up since it happens in another thread.